skimage/measure/tests/test_regionprops.py

```diff
@@ -184,12 +185,33 @@ def test_equiv_diameter():
 
 def test_euler_number():
     en = regionprops(SAMPLE)[0].euler_number
-    assert en == 1
+    assert en == 0
 
     SAMPLE_mod = SAMPLE.copy()
     SAMPLE_mod[7, -3] = 0
     en = regionprops(SAMPLE_mod)[0].euler_number
-    assert en == 0
+    assert en == -1
+
+    en = euler_number(SAMPLE, 4)
+    assert en == 2
+
+    en = euler_number(SAMPLE_mod, 4)
+    assert en == 1
+
+    en = euler_number(SAMPLE_3D, 6)
+    assert en == 1
+
+    en = euler_number(SAMPLE_3D, 26)
+    assert en == 1
+
+    SAMPLE_3D_2 = np.zeros((100, 100, 100))
+    SAMPLE_3D_2[40:60, 40:60, 40:60] = 1
+    en = euler_number(SAMPLE_3D_2, 26)
+    assert en == 1
```

Comment:
this has the topology of a ball, I would expect its Euler characteristic to be 2, cf https://en.wikipedia.org/wiki/Euler_characteristic

Reply:
In 3D, you add the number of faces. I admit I was also surprised of this result. 
To verify this, I used the matlab code from David Legland (Geometric measures in 2D/3D images):

```
% test euler 3D
I = zeros(100,100,100);
I(40:60, 40:60, 40:60) = 1;
imEuler3d(I)

I(45:55, 45:55, 45:55) = 0;
imEuler3d(I)
---
ans =
     1
ans =
     2
```

Reply:
In the wikipedia page, they refer to _the surfaces of polyhedra_ . For a convex body, the Euler number is 1. Have a look at the last section (generalization), the formula involves an alternating sum.

In 3D Images of Materials Structures, (Ohser and Schladitz, Wiley,2009), Eq. 2.4, you find a formula for polytopes I interpret as:
euler number = nb vertices - nb edges + nb faces - nb objects
which gives 1 for a non void cube.